controller/engine_controller.go

```diff
@@ -388,6 +388,20 @@ func (ec *EngineController) DeleteInstance(obj interface{}) error {
 		return nil
 	}
 
+	im, err := ec.ds.GetInstanceManager(e.Status.InstanceManagerName)
+	if err != nil {
+		return err
+	}
+
+	// Node down
+	if im.Spec.NodeID != im.Spec.OwnerID {
```

Comment:
The semantic of `im.Spec.NodeID != im.Spec.OwnerID` is not equal to `Node down`, it would depend on the implementation details. Do we have a more straightforward way to know the instance manager won't be able to respond due to the node is down?

Reply:
Use `ds.IsNodeDownOrDeleted(im.Spec.NodeID)`to double check it?

Reply:
Yeah, sounds better.